src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -2718,26 +1779,18 @@ bool processOnChangeConditionForSubscription
   const std::string&               tenant,
   const std::string&               xauthToken,
   const std::vector<std::string>&  servicePathV,
-  const std::string&               qFilter
+  Restriction*                     resP
```

Comment:
I think (not sure) that the convenion for read-only parameter was to use const X&. Thus, why don't use the following for this?

```
const Restriction& res
```

(In some cases, it is difficult for the caller to get the data as const X& and a pointer is better. However, please consider if it possible to do it in "the right way". If not possible, mark this as NTC).